instrumentation/java11-http-client/java-http-client/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpPayloadExtractor.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 Lumigo LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.javaagent.instrumentation.httpclient;
+
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Flow;
+import java.util.logging.Logger;
+
+public class HttpPayloadExtractor implements AttributesExtractor<HttpRequest, HttpResponse<?>> {
+  private static final Logger LOGGER = Logger.getLogger(HttpPayloadExtractor.class.getName());
+
+  private static final String HTTP_REQUEST_BODY_KEY = "http.request.body";
+  static final String HTTP_RESPONSE_BODY_KEY = "http.response.body";
+
+  @Override
+  public void onStart(AttributesBuilder attributes, Context parentContext, HttpRequest httpRequest) {
+    if (httpRequest.bodyPublisher().isEmpty()) {
+      // No HTTP Request BodyPublisher present
+      return;
+    }
+
+    HttpRequest.BodyPublisher bodyPublisher = httpRequest.bodyPublisher().get();
+    if (bodyPublisher.contentLength() != 0) {
+      // Subscribe to the HttpRequest.BodyPublisher and capture the content
+      RequestPayloadSubscriber requestPayloadSubscriber = new RequestPayloadSubscriber(attributes);
+      bodyPublisher.subscribe(requestPayloadSubscriber);
+    }
+  }
+
+  @Override
+  public void onEnd(
+      AttributesBuilder attributes,
+      Context context,
+      HttpRequest httpRequest,
+      HttpResponse httpResponse,
+      Throwable error) {
+
+    // Set the captured response payload onto a Span attribute
+    if (null != httpResponse && httpResponse.statusCode() == 200) {
```

Comment:
why does the response needs to be other than 200?

Reply:
We're only capturing response bodies when it's a 200.

If there's other status codes which could have a meaningful response body we can include them. This was mainly to prevent setting a `null` response body when there was no body, as `null` could be a valid response